pkg/odo/cli/utils/convert.go

```diff
@@ -0,0 +1,399 @@
+package utils
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/fatih/color"
+	imagev1 "github.com/openshift/api/image/v1"
+
+	"github.com/openshift/odo/pkg/config"
+	"github.com/openshift/odo/pkg/devfile/parser"
+	devfileCtx "github.com/openshift/odo/pkg/devfile/parser/context"
+	"github.com/openshift/odo/pkg/devfile/parser/data"
+	"github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/envinfo"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/occlient"
+	"github.com/openshift/odo/pkg/odo/cli/component"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+	"github.com/openshift/odo/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+)
+
+const (
+	convertCommandName = "convert-to-devfile"
+	//build command id to be used in s2i devfile
+	buildCommandID = "s2i-assemble"
+	// build command to be used in s2i devfile
+	buildCommandS2i = "/opt/odo/bin/s2i-setup && /opt/odo/bin/assemble-and-restart"
+	// run command id to be used in devfile
+	runCommandID = "s2i-run"
+	// run command to be used in s2i devfile
+	runCommandS2i = "/opt/odo/bin/run"
+	// container component name to be used in devfile
+	containerName = "s2i-builder"
+	// directory to sync s2i source code
+	sourceMappingS2i = "/tmp/projects"
+	// devfile version
+	devfileVersion = "2.0.0"
+	// environment variable set for s2i assemble and restart scripts
+	// some change in script if scripts is executed for a s2i component converted to devfile
+	envS2iConvertedDevfile = "ODO_S2I_CONVERTED_DEVFILE"
+)
+
+var convertLongDesc = ktemplates.LongDesc(`Converts odo specifice configuration from s2i to devfile, 
+It generates devfile.yaml and env.yaml for s2i components`)
+
+//var convertExample = ktemplates.Examples(`odo utils convert-to-devfile`)
+
+var convertExample = ktemplates.Examples(`  # Convert s2i component to devfile component
+
+Note: Run all commands from  s2i component context directory
+
+1. Generate devfile.yaml and env.yaml for s2i component.
+%[1]s  
+
+2. Push the devfile component to the cluster.
+odo push
+
+3. Verify if devfile component is deployed sucessfully.
+odo list
+
+4. Jump to 'rolling back migration', if devfile component deployment failed.
+
+5. Delete the s2i component.
+odo delete --s2i -a
+
+congratualtions, you have successfully converted s2i component to devfile component.
+
+# Rolling back the conversion
+1. If devfile component deployment failed, delete the devfile component, report this to odo dev community.
+odo delete -a
```

Comment:
So are they supposed to report to the dev community or run `odo delete -a` or both? Would `odo delete -a` delete both s2i & devfile configuration from the `.odo` directory? :thinking: